csharp/src/Apache.Arrow/Flatbuf/FlatBuffers/ByteBuffer.cs

```diff
@@ -833,7 +834,7 @@ public int Put<T>(int offset, T[] x)
                 AssertOffsetAndLength(offset, numBytes);
                 // if we are LE, just do a block copy
 #if ENABLE_SPAN_T
-                MemoryMarshal.Cast<T, byte>(x).CopyTo(_buffer.Span.Slice(offset, numBytes));
+                MemoryMarshal.Cast<T, byte>(x).CopyToFix(_buffer.Span.Slice(offset, numBytes));
```

Comment:
The Flatbuffer code is copied from https://github.com/google/flatbuffers/tree/master/net/FlatBuffers (which is the way https://google.github.io/flatbuffers/flatbuffers_guide_use_java_c-sharp.html says is how you consume flat buffer code).

I did a few checks, and the methods you are changing are not called in the Arrow library.

Can you leave the FlatBuffers code unchanged? That way when we update it from google/flatbuffers again, we don't have to worry about merge conflicts.

See https://github.com/apache/arrow/tree/master/csharp#updating-flatbuffers-code for more info.